doc-content/drools-docs/src/main/asciidoc/DecisionEngine-chapter.adoc

```diff
@@ -0,0 +1,55 @@
+[[drools.DecisionEngine]]
+
+= {DECISION_ENGINE}
+
+As a business rules developer, your understanding of the {DECISION_ENGINE} can help you design more effective business assets in {PRODUCT} and a more scalable decision management architecture. The {DECISION_ENGINE} is the {PRODUCT} component that stores, processes, and evaluates data to execute business rules and to reach the decisions that you define. This chapter describes basic concepts and functions of the {DECISION_ENGINE} to consider as you create your business rule system and decision services in {PRODUCT}.
+
+include::DecisionEngine/phreak-algorithm-con.adoc[leveloffset=+1]
```

Comment:
I'm not sure I follow. Do you mean this one parent section/paragraph, or the entire document? This one section is the parent introductory section for all the rest of the decision engine content about the decision engine, which is why it's first. And I have the all the content within it as chapter 3 in the Drools community doc, following Introduction and KIE. This is just following suit with the structure of the overall doc (overview of core components and behavior, then off to the rest about working with rules, etc.).

Sure, a generic section like that would be most welcome! One on rules engines in general, facts, rules, KIE base, working memory, maybe KIE module stuff, etc. What little we do have for these are documented in the [packaging and deploying ](https://access.redhat.com/documentation/en-us/red_hat_process_automation_manager/7.2/html-single/packaging_and_deploying_a_red_hat_process_automation_manager_project/index#project-kmodule-configure-proc_packaging-deploying)enterprise doc and in the r[ule language reference section](http://file.rdu.redhat.com/~sterobin/BXMSDOC-3978_COMMUNITY/#_droolslanguagereferencechapter) of community. Since this doesn't really exist yet, it would be easier if you or Luca or Mario wrote it and then I could work with you on it from there. Up to you. Could also consider it for 7.5, because I have to come back to this to add some content from the [running chapter](http://file.rdu.redhat.com/~sterobin/BXMSDOC-3978_COMMUNITY/#_running) of Drools at that point as well.

Reply:
no, I just meant Chapter 1. I think keeping the algo here is too sudden. It made more sense in the original documentation (which I still wouldn't organize the way it is, but that's for another day :P) because it was more contextualized, but if we do want to keep this information, I'd put it in an appendix. Instead I'd rather add a veeeery short introduction to the terms we are going to use in the following and a general overview of what a rule engine does. Just enough to introduce the way the product works (a synthesis of 3.1.3-3.1.4 community docs will probably work).

We can certainly take a look at this if there is no rush.

Reply:
@evacchi, I've just added this new chapter 1: [Decision engine in DM/PAM](http://file.rdu.redhat.com/~sterobin/BXMSDOC-3978_DM/#decision-engine-con_decision-engine). Same in [Drools](http://file.rdu.redhat.com/~sterobin/BXMSDOC-3978_COMMUNITY/#decision-engine-con_decision-engine), but conditionalized.

Is this new section correct and complete, and does it satisfy what you were hoping for? The diagram will be corrected in the redesigned version, as we discussed.

Reply:
looks good to me @mariofusco @lucamolteni can either of you double check?